ciao-controller/instance.go

```diff
@@ -63,17 +64,23 @@ func isCNCIWorkload(workload *types.Workload) bool {
 }
 
 func newInstance(ctl *controller, tenantID string, workload *types.Workload,
-	volumes []storage.BlockDevice, name string) (*instance, error) {
+	volumes []storage.BlockDevice, name string, subnet string) (*instance, error) {
 	id := uuid.Generate()
 
 	if name != "" {
-		existingID, err := ctl.ds.ResolveInstance(tenantID, name)
+		instance, err := ctl.ds.ResolveInstance(tenantID, name)
 		if err != nil {
 			return nil, errors.Wrap(err, "error trying to resolve name")
 		}
 
-		if existingID != "" {
-			return nil, fmt.Errorf("Instance name already in use: %s", name)
+		if instance != nil {
+			// allow repeat names on stopping instances
+			instance.StateLock.RLock()
+			if instance.State != payloads.Stopping {
```

Comment:
@rbradford I don't think it's appropriate to do the check inside ResolveInstance, and so the only other thing we could do is make some kind of random name for the cnci. However, I think it's perfectly fine to have duplicate names when an instance is about to stop anyway.